ddtrace/tracer/tracer.go

```diff
@@ -356,8 +418,8 @@ func (t *tracer) worker(tick <-chan time.Time) {
 		select {
 		case trace := <-t.out:
 			t.sampleChunk(trace)
-			if len(trace.spans) != 0 {
-				t.traceWriter.add(trace.spans)
+			if len(trace.Spans) != 0 {
+				t.traceWriter.add(trace.Spans)
 			}
```

Comment:
Feels to me that `t.traceWriter.add` should have no side-effect if passed an empty slice, which would make this guard here unnecessary?